docs/DeveloperGuide.md

```diff
@@ -1,38 +1,280 @@
 # Developer Guide
 
+## Table of Contents
+
++ [Acknowledgements](#acknowledgements)
++ [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#ui-component)
+    * [Storage Component](#storage-component)
++ [Implementation](#implementation)
++ [Product Scope](#product-scope)
+    * [Target User Profile](#target-user-profile)
+    * [Value Proposition](#value-proposition)
++ [User Stories](#user-stories)
++ [Non-Functional Requirements](#non-functional-requirements)
++ [Glossary](#glossary)
++ [Instructions for Manual Testing](#instructions-for-manual-testing)
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+## Design
+
+### Architecture
+
+### Parser Component
+The `Parser` class handles the parsing of user input, executing the appropriate tasks
+or throwing relevant errors.
+* `Parser` first receives user input from `Duke` through `Parser::handleCmd`
+* Depending on the command given, `Parser` will then execute the command on the `ExpenseManager`instance, or throw a `DukeException`, which will be caught by `Duke`
+### **Sequence Diagram**
+<img alt="parser sequence diagram" src="diagrams/parser.png" width="280" />
```

Comment:
The names of the classes need a colon at the beginning. Ex -- :Parser, :Duke, :ExpenseManager